src/basic/time-util.c

```diff
@@ -1311,6 +1311,9 @@ bool timezone_is_valid(const char *name, int log_level) {
         if (slash)
                 return false;
 
+        if (p - name >= PATH_MAX)
+                return false;
```

Comment:
This check *could* moved into the loop, but since this is only for pathological cases, it doesn't really matter.